### PR TITLE
from_torch  conversion should rely on device Ops

### PR DESCRIPTION
**Pull request by aliaksei-sala**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/24217)

### Problem description
This PR was created to facilitate merging the following PR from the fork: https://github.com/tenstorrent/tt-metal/pull/24521

### What's changed
Prepared the infrastructure for offloading computation to the device. The device operations are currently disabled due to several issues, including precision loss and unit test failures (for example, incorrect cache entry counts). Enabling device-side operations will be addressed in a follow-up PR.

### Checklist

- [x] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml?query=branch:mirror/maxim_artemov/TTNN-24217-torch-conversion-via-typecast)
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:mirror/maxim_artemov/TTNN-24217-torch-conversion-via-typecast)
- [ ] [cpp-unit-tests](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:mirror/maxim_artemov/TTNN-24217-torch-conversion-via-typecast)
- [x] [T3K Demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml?query=branch:mirror/maxim_artemov/TTNN-24217-torch-conversion-via-typecast) 
Some of the model performance results vary between runs,  t3k_qwen3_tests is passing [here](https://github.com/tenstorrent/tt-metal/actions/runs/21192638127)
[Main T3K Demo tests](https://github.com/tenstorrent/tt-metal/actions/runs/21419668464/job/61676859653)
- [x] [Galaxy Deepseek tests](https://github.com/tenstorrent/tt-metal/actions/workflows//galaxy-deepseek-tests.yaml?query=branch:mirror/maxim_artemov/TTNN-24217-torch-conversion-via-typecast) 

#### Model tests

If your changes cover model-related code, you should run tests corresponding to affected models and platforms (Single card, T3K, Galaxy). "Choose your pipeline" workflows facilitate running multiple kinds of tests in a single run. Each offers `models-mandatory` and `models-extended` presets.
The former includes a minimal set of tests, to be run always. The latter extends that with additional ones - use your best judgement in deciding which is the most appropriate for your PR.

- [x] [(Single) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:mirror/maxim_artemov/TTNN-24217-torch-conversion-via-typecast). (No regression)
  - [ ] `models-mandatory` preset (runs: [Device perf regressions](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) and [Frequent model and ttnn tests](https://github.com/tenstorrent/tt-metal/actions/workflows/fast-dispatch-full-regressions-and-models.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) tests)
  - [ ] other selection - specify runs

- [ ] [(T3K) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:mirror/maxim_artemov/TTNN-24217-torch-conversion-via-typecast)
  - [ ] `models-mandatory` preset (runs: [Unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-model-perf-tests.yaml) tests)
  - [ ] other selection - specify runs

- [ ] [(Galaxy) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:mirror/maxim_artemov/TTNN-24217-torch-conversion-via-typecast)
  - [ ] `models-mandatory` preset (runs: [Quick tests](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-model-perf-tests.yaml) tests)
  - [ ] other selection - specify runs

### CODE REVIEW

**Comment by blozano-tt** [comment]
Added bypass, because this PR is mirror of: https://github.com/tenstorrent/tt-metal/pull/24521

File: tests/ttnn/unit_tests/test_torch_conversion.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+from tests.ttnn.utils_for_testing import assert_with_pcc, comp_pcc, tt_dtype_to_torch_dtype
+import torch
+import ttnn
+import pytest
+import numpy as np
+
+
+def is_ttnn_float_type(tt_dtype) -> bool:
```
[the remark this answers is not in the document]
Comment (aliaksei-sala) [reply tests/ttnn/unit_tests/test_torch_conversion.py:12]: Adding `return False` at the end of the function would trigger an “unreachable statement” error, so we should keep the current implementation.
